x-pack/plugins/security_solution/scripts/openapi/generate.js

```diff
@@ -9,10 +9,10 @@ require('../../../../../src/setup_node_env');
 const { generate } = require('@kbn/openapi-generator');
 const { resolve } = require('path');
 
-const SECURITY_SOLUTION_ROOT = resolve(__dirname, '../..');
+const X_PACK_PLUGINS_ROOT = resolve(__dirname, '../../..');
 
 generate({
-  rootDir: SECURITY_SOLUTION_ROOT,
-  sourceGlob: './**/*.schema.yaml',
+  rootDir: X_PACK_PLUGINS_ROOT,
+  sourceGlobs: ['./security_solution/**/*.schema.yaml', './osquery/**/*.schema.yaml'],
   templateName: 'zod_operation_schema',
 });
```

Comment:
Hey @tomsonpl 👋 It's great to see you've started using the code generator for Osquery!

I wanted to clarify the intended use of the `@kbn/openapi-generator` package, as it may not be entirely obvious. We built it for reuse and customization, allowing different plugin owners to configure code generation targets for their specific needs. For example, the Security Solution team doesn't need to know the generation targets and folder structures for other plugins, and vice versa. We're also planning to add more templates and targets in the Security Solution that will be specific to our use cases, which may not be relevant for other plugins.

So, my suggestion is to separate our codegen scripts. Here’s how you can do it:
1. Create a script for Osquery similar to the one we use. You can find more details here: https://github.com/elastic/kibana/blob/main/packages/kbn-openapi-generator/README.md#programmatic-api. Or, for simplicity, you could use the CLI and run `node scripts/generate_openapi --rootDir ./x-pack/plugins/osquery` from the Kibana root.
2. Set up CI integration for Osquery following these instructions: https://github.com/elastic/kibana/blob/main/packages/kbn-openapi-generator/README.md#ci-integration

This approach will give us two independent sets of code generation scripts. If you have any questions, I’ll be happy to jump on a call.

Reply:
Ok that makes sense, will adjust my PR :) Thanks :)